packages/dev/test/RelayHub.test.ts

```diff
@@ -772,6 +775,56 @@ contract('RelayHub', function ([_, relayOwner, relayManager, relayWorker, sender
             expectEvent.inLogs(logs, 'TransactionRelayed', { status: RelayCallStatusCodes.PaymasterBalanceChanged })
           }
         })
+        context('with BatchGateway configured', function () {
+          const batchGateway = other
+
+          let gatewayForwarder: GatewayForwarderInstance
+          let relayHubInstance: RelayHubInstance
+          let recipientContract: TestRecipientInstance
+          let relayRequest: RelayRequest
+
+          beforeEach(async function () {
+            relayRequest = cloneRelayRequest(sharedRelayRequestData)
+            gatewayForwarder = await GatewayForwarder.new()
```

Comment:
any special reason to construct all contracts before each test?

Reply:
(that is "before" is enough)

Reply:
There is a weird order the framework calls 'before/beforeEach' in a nested test. I could not figure it out but the inner 'beforeEach' is run after the outer 'beforeEach', which is something I do need here.

Reply:
I don't think its weird: it should always run the "before" in order from outer to inner, and the "after" in the inner-to-outer order.
do you need anything else?